src/prommis/uky/costing/ree_plant_capcost.py

```diff
@@ -2669,3 +2858,737 @@ def costing_upper_bound_eq(c, p):
+            doc="amortized annual payment on loans in $MM",
+            units=b.cost_units,
+        )
+
+        b.pv_loan_interest = Var(
+            initialize=-b.CAPEX,
+            bounds=(-1e4, 0),
+            doc="present value of total lifetime loan interest in $MM; negative cash flow",
+            units=b.cost_units,
+        )
+
+        b.pv_capital_depreciation = Var(
+            initialize=-b.CAPEX,
+            bounds=(-1e4, 0),
+            doc="present value of total lifetime capital depreciation in $MM; negative cash flow",
+            units=b.cost_units,
+        )
+
+        b.npv = Var(
+            initialize=-b.CAPEX,
+            bounds=(-1e4, 1e4),
+            doc="present value of total lifetime capital depreciation in $MM",
+            units=b.cost_units,
+        )
+
+        # build parameters
+
+        b.discount_percentage = Param(
+            initialize=b.config.discount_percentage, units=pyunits.percent
+        )
+        b.plant_lifetime = Param(
+            initialize=b.config.plant_lifetime, units=pyunits.years
+        )
+
+        if b.config.has_capital_expenditure_period:
+            b.capital_expenditure_percentages = Param(
+                range(len(b.config.capital_expenditure_percentages)),
+                initialize=dict(
+                    zip(
+                        range(len(b.config.capital_expenditure_percentages)),
+                        b.config.capital_expenditure_percentages,
+                    )
+                ),
+            )
+
+        b.capital_escalation_percentage = Param(
+            initialize=b.config.capital_escalation_percentage, units=pyunits.percent
+        )
+
+        b.operating_inflation_percentage = Param(
+            initialize=b.config.operating_inflation_percentage, units=pyunits.percent
+        )
+        b.revenue_inflation_percentage = Param(
+            initialize=b.config.revenue_inflation_percentage, units=pyunits.percent
+        )
+
+        b.royalty_charge_percentage_of_revenue = Param(
+            initialize=b.config.royalty_charge_percentage_of_revenue,
+            units=pyunits.percent,
+        )
+
+        b.debt_percentage_of_CAPEX = Param(
+            initialize=b.config.debt_percentage_of_CAPEX, units=pyunits.percent
+        )
+        b.loan_interest_percentage = Param(
+            initialize=b.config.loan_interest_percentage, units=pyunits.percent
+        )
+        b.loan_repayment_period = Param(
+            initialize=b.config.loan_repayment_period, units=pyunits.years
+        )
+
+        b.capital_depreciation_declining_balance_percentage = Param(
+            initialize=b.config.capital_depreciation_declining_balance_percentage,
+            units=pyunits.percent,
+        )
+
+        # define series present worth factor as an method so it can be called
+
+        def series_present_worth_factor(r, g, N):
+            """
+            Returns expression for series present worth factor.
+            """
```

Comment:
```suggestion
            """
            Returns expression for series present worth factor where r is the discount rate
            expressed as a decimal, N is the project lifetime, and g is the escalation rate
            (e.g. inflation) expressed as a decimal.
            """
```
It might be worth adding this tidbit here as well for convenience